lib/awful/titlebar.lua

```diff
@@ -513,6 +513,8 @@ local function new(c, args)
     -- Handle declarative/recursive widget container
     ret.setup = base.widget.setup
 
+    c.titlebars = bars
```

Comment:
Where is this used? I don't think I saw any use in this PR, so just drop this commit from this PR.

Reply:
It is used in the awful/template.lua

Reply:
Ah, some already-merged commit uses this. Okay, so I'll have to take a look at how this stuff works... The drawin shim doesn't actually do anything and certainly cannot be drawn to.... Can't you do something similar to baf290a8b22560d3d860bb75ec1429442417093b in `wibox.drawable`? I.e. add a `d.get_wibox_drawable()` function in there? That way the shim client can keep a reference to the titlebar drawin that it returns, the drawin has a reference to its drawable, the new function allows you to get the wibox.drawable from this and this is (basically) all that `wrap_titlebar()` from `awful/template.lua` needs (widget, bg and fg color).

Thus, instead of making this implementation detail of `awful.titlebar` public, we just need "something similar" in `wibox.drawable` to what you already did in `wibox`.

What do you think?

Reply:
> so I'll have to take a look at how this stuff works...

Ping?

Please note that it  is a private API. It can be changed later. Maybe it's better to keep it that way given #650 might cause us to break the low level titlebar API once either you or me decide to complete the implementation. Also, it's blocking the `popup` PR progress and, in turn, the v4.3 release.

Reply:
The `...` was meant to be the time that I used for looking at the shim's code. I did not actually mean something like "tomorrow".

Well, undocumented does not mean that people will not use it. And if the name does not start with e.g. `_private`, I wouldn't even really blame them for that.

What do you think about my proposal above to instead have the shims actually track the titlebar drawables that they hand out, instead of later trying to reconstruct that relation? Is it really better, or is it mostly "meh, whatever"?

Reply:
Ping?